include/ProcessTools.h

```diff
@@ -45,25 +45,93 @@
-        Mezzanine::String MEZZ_LIB GetFileContents(const Mezzanine::String& Filename);
-    }// Testing
+        /// @brief The output to cout and cerr from the called process.
+        String ConsoleOutput;
+        /// @brief The code returned when the called process exited.
+        Integer ExitCode = EXIT_FAILURE;
+    };//CommandResult
+
+RESTORE_WARNING_STATE
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Output to String
+
+    /// @brief Launches a different process on the system.
+    /// @note ExecutablePath cannot be empty on Posix systems or this function will fail.
+    /// @param ExecutablePath The path to the executable to be invoked.
+    /// @param Command The command to attempt to run and direct its output.
+    /// @return Returns the ExitCode and Cout/Cerr output of the command that was run.
+    CommandResult MEZZ_LIB RunCommand(const StringView ExecutablePath, const StringView Command);
+    /// @brief Launches a different process on the system.
+    /// @remarks This function will interpret all of the text up until the first delimiter (space or tab)
+    /// to be the path to the executable to be launched. If that executable has a space in it's path, use
+    /// the two parameter version of this function instead.
+    /// @param Command The command to attempt to run and direct its output.
+    /// @return Returns the ExitCode and Cout/Cerr output of the command that was run.
+    CommandResult MEZZ_LIB RunCommand(const StringView Command);
+
+    /// @brief Launches a different process on the system and return only it's console output as a String.
+    /// @note ExecutablePath cannot be empty on Posix systems or this function will fail.
+    /// @remarks This function ignores the exit code of the command. If you want/need the exit code, use
+    /// @ref RunCommand instead.
+    /// @param ExecutablePath The path to the executable to be invoked.
+    /// @param Command The command to attempt to run and direct its output.
+    /// @return Returns a String containing the output of the specified command.
+    [[nodiscard]]
+    String MEZZ_LIB GetCommandOutput(const StringView ExecutablePath, const StringView Command);
+    /// @brief Launches a different process on the system and return only it's console output as a String.
+    /// @remarks This function ignores the exit code of the command. If you want/need the exit code, use
+    /// @ref RunCommand instead. @n@n
+    /// This function will interpret all of the text up until the first delimiter (space or tab) to be the
+    /// path to the executable to be launched. If that executable has a space in it's path, use the two
+    /// parameter version of this function instead.
+    /// @param Command The command to attempt to run and direct its output.
+    /// @return Returns a String containing the output of the specified command.
+    [[nodiscard]]
+    String MEZZ_LIB GetCommandOutput(const StringView Command);
```

Comment:
These seem totally superceded for `RunCommand`. Can we safely remove these?

If not safe, can we remove them?

Reply:
After chatting face to face, maybe we can mark this deprecated or yank it. I just think it is poor form encouraging ignoring the return code of the executable.

Reply:
These exist for convenience.  I'd imagine we could remove them entirely with little to no consequence, yes.

Reply:
For some reason I though a previous version. I don't think that`Testing::GetCommandOutput("cmake -E echo foo")).ConsolePut` is any easier or harder than a second function that does that. But I could see leaving it as well.